Fix incremental builds

ewanharris opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-28130

* Unmarks the Product directory to prevent us removing required files like the tiverify lib and TitaniumKit framework
* Small change to the buildModule file to call `Array.from` on the architectures set when logging it to  avoid it being logged as `[object Set]`

ewanharris committed:
fix(ios): unmark product directory to prevent removing required files

It's probable that this might be too much

Fixes TIMOB-28130

ewanharris committed:
fix(ios): convert architechture list to array before logging

Currently these are logged as [Object Set] as I'm guessing the underlying logger doesn't know how
to convert a set

build commented:
<!--
  0 failure: 
  0 warning: 
  2 messages
  
  DangerID: danger-id-Danger;
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: All tests are passing
Nice one! All 5083 tests are passing.
(There are 472 skipped tests not included in that total)
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d3dd1678317912843fc82e9e79216909e8297427
</p>

ewanharris committed:
fix(ios): remove Frameworks directory prior to copying

If any framework underneath changes (say the appc-cli is injecting the appc-verify pieces) then it
will cause the Frameworks to be copied again and as there are symlinks the copy will get funky
and try to place symlinks underneath each other. Removing the directory before copying just
prevents this error from happening

sgtcoolguy commented on iphone/cli/commands/_build.js:
```diff
@@ -4638,7 +4638,7 @@ iOSBuilder.prototype.copyTitaniumiOSFiles = function copyTitaniumiOSFiles() {
 						// Copy whole 'Frameworks' directory from SDK to build directory, to preserve symlink available in Titaniumkit.xcframework.
 						// TODO: Is there any better way to do this?
-
+						fs.emptyDirSync(path.join(this.buildDir, dir));
```
may want to stick a comment in the code why we're doing this so it doesn't get reverted way down the road.

ewanharris replied:
Had missed this comment, just pushed up a commit now that explains why we nuke before copying

sgtcoolguy reviewed (approved):
LGTM

cb1kenobi reviewed (approved):
LGTM

ewanharris committed:
refactor(ios): add comment on why we're removing the frameworks directory on change

ssjsamir reviewed (approved):
FR Passed: Able to build incrementally without any errors (iOS and MacOS)

**Test Environment**
```
MacOS Big Sur: 11.0 Beta 6
Xcode: 12.0 Beta 5
Java Version: 1.8.0_242
Android NDK: 21.3.6528147
Node.js: 12.18.1
""NPM":"5.0.0","CLI":"8.1.1-master.5""
iphone 8 Sim (14.0 Beta)
```
